docs/design/tanka.md

```diff
@@ -0,0 +1,205 @@
+# tanka, advanced Kubernetes configuration
+**Author**: @sh0rez  
+**Date**: 26.07.2019
+
+## Motivation
+`json` and it's superset `yaml` are great configuration languages for
+machines, but they are [not really good for humans](https://youtu.be/FjdS21McgpE).
+
+But the Kubernetes ecosystem seems to have settled to configuring all kinds of
+workloads using exactly these. While they provide a low entry barrier, it is
+challenging to model advanced requirements, for example deploying the same
+application to multiple clusters or to different environments.
+
+This usually leads to duplication on two level: The obvious one is the
+application level. Once an app needs to redeployed to another environment
+(`dev`, `prod`, etc.), it usually shares most of the configuration, except for
+some edge-cases (secrets, etc). `kubectl` does not really provide a utility to
+fully address this.
+
+Even more duplication happens on the systems level. Today, multiple applications
+are usually composed into a larger picture.  
+But deploying common building blocks as `postgresql` or `nginx` requires the
+same code (`Deployment`, `Service`, etc) that nearly everyone attempting to use
+these will write.
+
+While maintaining the same code for `dev` and `prod` might go well, it becomes
+unconquerable once it comes to multi-region (5+) deployments or even more
+versatile use-cases.
+
+#### Existing solutions
+Historically, this problem has been approached using string-templating (`helm`).
+While `helm` allows code-reuse using `values.yml`, a chart is only able to provide what
+has been thought of during writing. Even worse, charts are maintained inside of
+the `helm/charts` repository, which makes quick or even domain-specific edits
+hard. It is impossible to easily address edge-cases.
+
+A solution to this problem is called [jsonnet](https://jsonnet.org): It is
+basically `json` but with variables, conditionals, arithmetic, functions,
+**imports**, and error propagation and especially very clever **deep-merging**.
```

Comment:
Can we elaborate on what deep-merging is? Maybe in an `Appendix` section?

Reply:
Is the example in [Edge Cases](https://github.com/sh0rez/tanka/blob/095f5d535b99181ec1f71b41f2bf6070ff6379f6/docs/design/tanka.md#edge-cases) enough? We could link there